crates/rust-analyzer/src/reload.rs

```diff
@@ -139,9 +139,10 @@ impl GlobalState {
                 "Proc-macros and/or build scripts have changed and need to be rebuilt.\n\n",
             );
         }
-        if self.fetch_build_data_error().is_err() {
-            status.health |= lsp_ext::Health::Warning;
+        if let Err(build_data_err) = self.fetch_build_data_error() {
+            status.health |= lsp_ext::Health::Error;
             message.push_str("Failed to run build scripts of some packages.\n\n");
+            format_to!(message, "{build_data_err}\n");
```

Comment:
The issue with this is I believe that we have limited real estate in the hover box for the status message and so depending on the error it might not fit. We are logging the errror though so maybe we should instruct the user to look at the logs